incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -38,33 +45,50 @@ type ConfigReconciler struct {
 // Reconcile sets up some basic variable and logs the Spec.
 // TODO: Updates the comment above when adding more logic to the Reconcile method.
 func (r *ConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	// TODO: Surface the error more prominently and add a validating admission controller to prevent
-	// the problem in the first place.
+	ctx := context.Background()
+
+	// Validate the singleton name.
+	// TODO: Add a validating admission controller to prevent the problem in the first place.
 	if req.NamespacedName.Name != api.HNCConfigSingleton {
-		r.Log.Error(nil, "Singleton name is wrong. It should be 'config'.")
+		nnm := types.NamespacedName{Name: req.NamespacedName.Name}
```

Comment:
Rather than reading/writing the "bad" config, maybe let's just add a condition to the "real" config and leave the bad one alone? That seems like a lot less code.

Reply:
I feel putting the error in the real config might be confusing: the singleton has the correct name "config" but has the error that ""Singleton name is wrong. It should be 'config'". 

Furthermore, when people create the object with a wrong name, they probably don't know the "config" object exists. If something is wrong, they are more likely to lookup the status of the object that they created to see what is going on instead of the status of the real "config" object.

Feel free to let me know your thoughts. :)

Reply:
Yeah, maybe both are needed. Though I realize that if someone adds the "bad" config, we'll have to manually re-enqueue the good one too - the "ignite" channel can be used for that though.

On the good one, you can use a more descriptive error like `The HNC Config must be called "config", but an HNC Config object with the name "foo" exists as well.`